packages/react/src/prop-types/deprecateValuesWithin.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { warning } from '../internal/warning';
+
+const didWarnAboutDeprecation = {};
+
+export default function deprecateValuesWithin(
+  propType,
+  message,
+  allowedValues
+) {
+  return function checker(props, propName, componentName, ...rest) {
+    if (props[propName] === undefined) {
+      return;
+    }
+
+    if (
+      !didWarnAboutDeprecation[componentName] ||
+      !didWarnAboutDeprecation[componentName][propName]
+    ) {
+      didWarnAboutDeprecation[componentName] = {
+        ...didWarnAboutDeprecation[componentName],
+        [propName]: true,
+      };
+
+      if (allowedValues && !allowedValues.includes(props[propName])) {
+        warning(
+          false,
+          message ||
+            `The value for \`${propName}\` prop has been deprecated, It will be removed in the next major release. Allowed values is/are: ${allowedValues.join(
```

Comment:
Can we add component name as well in warn message.

Reply:
Sure! Updated the same, also this string will only be displayed when `message` argument is not passed.